### PR TITLE
Update Card radius

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
Update the border radius applied to the `Card` component.

## Why?
https://github.com/WordPress/gutenberg/pull/64368 sought to update all components, but Card was missed. This PR rectifies that.

`radiusLarge` is applied as this aligns with the guidelines in the radius scale:

> `$radius-large: 8px;     // Applied to containers with larger padding.`

I think we can consider `Card` similar to `Modal` in this respect.

| Before | After |
| --- | --- |
| Screenshot 2024-09-04 at 13 49 46 | Screenshot 2024-09-04 at 13 56 49 |

## Visual changes in the editor
There are very few consumers of this component so visual changes are minimal. The main one seems to be the variation tile in global styles. 

Screenshot 2024-09-04 at 15 33 27

I'm not entirely sure the use of `Card` is warranted in this case, so this might be a detail to revisit separately. 

## Testing Instructions
1. `npm run storybook:dev`
2. Observe the updated radius applied to the Card component

### CODE REVIEW

File: packages/components/src/card/styles.ts
Comment (ciampo) [inline packages/components/src/card/styles.ts:15]: Is this something that we want to keep, or should we switch to `radiusMedium` ? 

Since we're defining a list of radii, I feel like we could try to avoid these exceptions.

**Review by ciampo — COMMENTED** [review]
Noting that we'll need to update snapshots and add a CHANGELOG entry for CI checks to pass

File: packages/components/src/card/styles.ts
Comment (jameskoster) [reply packages/components/src/card/styles.ts:15]: By 'this' do you mean the manual adjustment?

**Comment by jasmussen** [comment]
For the style card use case, 8px feels a bit large doesn't it? Wouldn't 4px be more right in this case? I.e. wouldn't that card eventually be shown in context of other 4px radii?

File: packages/components/src/card/styles.ts
Comment (ciampo) [reply packages/components/src/card/styles.ts:15]: Yes, sorry if I was unclear. I mean the `border - 1px` calculation, instead of using a `radius` from the list of shared variables.

Comment (tyxla) [reply packages/components/src/card/styles.ts:15]: Or it might be a good idea to just change it to `radiusMedium` as both @ciampo and @jasmussen suggested, and ditch the calculation altogether.

Comment (jameskoster) [reply packages/components/src/card/styles.ts:15]: Without the calculation there is an issue with the full-bleed-content version. Note the tiny white space visible between the border and the red area:

Screenshot 2024-09-04 at 15 48 46

**Comment by jameskoster** [comment]
Conceptually Cards should be used to provide structure in more complex views by grouping related UI together in order to aid visual scanning. Imagine a settings page with several sections/sub-sections. Generally a Card would be quite a large container which suggests the large radius is warranted, according to the scale.

I agree in the case of the style tile that `8px` seems too large, but I'm not entirely sure that `Card` is the right component to use there. I noticed there seems to be a bug; hovering the tile changes the cursor to `pointer` but there doesn't seem to be any interactivity. Should this element be refactored into a button that takes you to browse styles?

Alternatively, if we don't want to add any interactivity it could just be a plain container, not a Card. Finally, if we want to continue using Card but don't want the roundness, we could just apply `isRounded: false`.

**Comment by jasmussen** [comment]
I'm biased against cards, and in favor of spacing and borders to place out things instead. But since the component exists, absolutely important to make it shine, potentially even elevation can play a role in the card, i.e. if it's elevation 0 it doesn't have a border? 

About the radius, though, would you ever use a card inside a modal? If yes, then the radius should be smaller than the modal no? 

If you feel strongly about the 8px radius, I wouldn't mind it, though I would find a different component to use to wrap the style card, so that maintains the same radius as its immediate surroundings.

File: packages/components/src/card/styles.ts
Comment (ciampo) [reply packages/components/src/card/styles.ts:15]: I had a look at the code, and I now understand the `-1px`. Since the border is rendered via `box-shadow`, it will implicitly increase by `1px` with respect to the `border-radius` — removing `1px` will offset that, making sure that whatever border radius we pick, it will be applied as expected.

In short, for the scope of this PR, we should be good with the `-1px`

**Comment by ciampo** [comment]
> hovering the tile changes the cursor to `pointer` but there doesn't seem to be any interactivity. Should this element be refactored into a button that takes you to browse styles?

@jameskoster , can you reproduce this behaviour in Storybook? If not, it may be something related to a specific use of `Card`

---

I see that we are discussing `Card` 's role in Gutenberg's UI (and therefore, the best border radius for it) — here are my thoughts:

- We have a `Surface` component that I don't think is particularly used — we should also consider whether is makes sense to keep it around (it may become more useful with theming, in order to get the correct background + text + border colors?), or otherwise consider deprecating it;
- same considerations for `Card`. If we want to keep it around, it may be a good time to review the component's design and its props/behaviour:
  - does it still make sense to use box-shadow for the borders?
  - the component seems to render double borders if `isBorderless = false` and `borderLeft / borderTop / borderRight / borderBottom` are also set to `true` (sidenote: do we need to control borders to this fine-grain level?)
  - we may want to review the `size` prop, since it may be a bit misleading (it only changes the padding, not the actual size of the card)
  - we may want to review the if the different `variant`s offerered are still relevant with the current design specs
  - we may want to review the elevation too

**Comment by jameskoster** [comment]
> About the radius, though, would you ever use a card inside a modal? If yes, then the radius should be smaller than the modal no?

It's hard to say for sure, but my instinct is no. Modals/Dialogs should _generally_ be concise interfaces that do not require the additional structure that cards provide. If a designer ends up using a Card inside a Modal then I'd be inclined to question whether a Modal is the right interface. 

One good thing is that Card isn't used much currently, and this is a tiny visual change in the grand scheme, so we can easily revisit later if/when consumption sees an uptick and we have a better idea about how it's being used.

@ciampo these are all good considerations. I agree the design/implementation could be revisited, but until there's an immediate need it's unlikely to be prioritised. This PR is mostly about updating the current implementation to adopt the design system fundamentals. 

As I see it the path forward could be:

1. Update the border radius applied to Card.
2. Revisit the implementation of the Style tile in Global Styles, likely to use a different component or just plain markup.
3. Revisit the broader spec for Card as the need arises.

If we want to avoid any temporary visual diffs then step 2 should happen in this PR, or a separate one that is merged before this one. Any strong feelings there?

---

All that said, I don't see this as a huge priority. If we'd rather leave it as-is for now and update the entire component later that also works for me.

**Comment by ciampo** [comment]
Sounds good, definitely a good idea to work on this iteratively and when there's a higher priority. I was just highlighting that there's an opportunity to make changes to the components while we're taking a look at the design system.

Once there's agreement on the actual border radius value, we can merge an iterate as appropriate 👌

**Comment by jameskoster** [comment]
To unblock this I've pushed a couple of small overrides in the site editor to keep the global styles UI closer to trunk.

| Trunk | This PR |
| --- | --- |
| Screenshot 2024-09-09 at 11 05 45 | Screenshot 2024-09-09 at 11 05 15 |

I still think we should revisit using `Card` here, but that's a separate discussion. I included a code comment to that effect.

File: packages/edit-site/src/components/global-styles/style.scss
Comment (ciampo) [inline packages/edit-site/src/components/global-styles/style.scss]: is the `!important` necessary?

**Review by ciampo — APPROVED** [review]
Code changes LGTM on my end — let's wait for Joen's reply re. the global styles override. Snapshots will need an update before merging.

File: packages/edit-site/src/components/global-styles/style.scss
Comment (jameskoster) [reply packages/edit-site/src/components/global-styles/style.scss]: The component styles are added inline, so unless there's a better way to perform the override (?) I think it is necessary.

**Comment by jameskoster** [comment]
What do you think @jasmussen ?

**Comment by jasmussen** [comment]
Sorry I missed the pings! 

I've longer term questions about the use of the card component itself, but none that are at all blocking to this work. Continue apace! (Knowing that the Style card was updated to not have the larger radius.)

File: packages/edit-site/src/components/global-styles/style.scss
Comment (ciampo) [reply packages/edit-site/src/components/global-styles/style.scss]: I don't see inline styles on those components — we can still apply overrides with a little specificity hack that is not as invasive as `!important`:

```scss
.edit-site-global-styles-screen-root__active-style-tile {
	// Todo: revisit use of Card component and/or remove override
	// The &#{&} is a workaround for the specificity of the Card component.
	&#{&},
	&#{&} .edit-site-global-styles-screen-root__active-style-tile-preview {
		border-radius: $radius-small;
	}
}
```

Comment (jameskoster) [reply packages/edit-site/src/components/global-styles/style.scss]: Sorry I meant added by js (I think). TIL there's a way to override that :)

**Comment by jameskoster** [comment]
@ciampo I think the snapshots need updating here. Would you mind handling that, or letting me know the command to do it myself? :D

File: packages/edit-site/src/components/global-styles/style.scss
Comment (ciampo) [reply packages/edit-site/src/components/global-styles/style.scss]: Yup! It's basically repeating the classname selector, ie. `.edit-site-global-styles-screen-root__active-style-tile.edit-site-global-styles-screen-root__active-style-tile`

**Comment by ciampo** [comment]
Sure! I rebased and pushed a commit to update the snapshots

FYI, `npm run test:unit -- --updateSnapshot` does it for me.

**Pull request merged**
